addons/website/models/website.py

```diff
@@ -1657,30 +1657,71 @@ def _trigram_enumerate_words(self, search_details, search, limit):
             fields = set(fields).intersection(model._fields)
 
             unaccent = get_unaccent_sql_wrapper(self.env.cr)
-            similarities = [sql.SQL("word_similarity({search}, {field})").format(
-                search=unaccent(sql.Placeholder('search')),
-                # Specific handling for website.page that inherits its arch_db and name fields
-                # TODO make more generic
-                field=unaccent(sql.SQL("{table}.{field}").format(
-                    table=sql.Identifier((self.env['ir.ui.view'] if (field in ['name', 'arch_db'] and model_name == 'website.page') else model)._table),
-                    field=sql.Identifier(field)
-                ))
-            ) for field in fields]
+
+            # Specific handling for fields being actually part of another model
+            # through the `inherits` mechanism.
+            # It gets the list of fields requested to search upon and that are
+            # actually not part of the requested model itself but part of a
+            # `inherits` model:
+            #     {
+            #       'name': {
+            #           'table': 'ir_ui_view',
+            #           'fname': 'view_id',
+            #       },
+            #       'url': {
+            #           'table': 'ir_ui_view',
+            #           'fname': 'view_id',
+            #       },
+            #       'another_field': {
+            #           'table': 'another_table',
+            #           'fname': 'record_id',
+            #       },
+            #     }
+            inherits_fields = {
+                inherits_model_fname: {
+                    'table': self.env[inherits_model_name]._table,
+                    'fname': inherits_field_name,
+                }
+                for inherits_model_name, inherits_field_name in model._inherits.items()
+                for inherits_model_fname in self.env[inherits_model_name]._fields.keys()
+                if inherits_model_fname in fields
+            }
```

Comment:
Kinda difficult to read :grimacing: But I guess other versions of it are not much more readable, I guess I may just not be familiar with python notations :grimacing:

Reply:
I wrote it in multiple ways as it's indeed not easy to read.
I initially went with the field name as key but it wasn't really easier/harder/short/longer :

```py
{
   'name': {
       'table': 'ir_ui_view',
       'inherit_field': 'view_id',
    },
    'url': {
       'table': 'ir_ui_view',
       'inherit_field': 'view_id',
    },
    'another_field': {
       'table': 'another_table',
       'inherit_field': 'record_id',
    }
}
```
But it contained duplicates that I had to merge in the second part of the fix below (left join part), so all in all I went with the current one.

I hesitated commenting every for loop to illustrate what are they returning, but at the end just did it through the generic comment above.

Note that this is greatly inspired by https://github.com/odoo/odoo/blame/8e4b9b9f1511096ce4a3ccd6f13044ce621c4b0b/odoo/models.py#L2911-L2916 (which was obviously shorter as no condition and just a tuple construction)

There is surely a better way I didn't think of :innocent: